addons/resource/models/resource_resource.py

```diff
@@ -196,8 +196,8 @@ def _get_valid_work_intervals(self, start, end, calendars=None, compute_leaves=T
         for calendar in (calendars or []):
             calendar_resources[calendar] |= self.env['resource.resource']
         for calendar, resources in calendar_resources.items():
-            # for fully flexible resource, return the whole interval
-            if not calendar:
+            # For flexible calendars and fully flexible resources: -> return the whole interval
+            if not calendar or calendar.flexible_hours:
```

Comment:
I don't think it is expected.

We did that logic for resources with flexible working schedule. 🤔 

https://github.com/odoo/odoo/blob/18.0/addons/resource/models/resource_calendar.py#L382-L432

Reply:
In other words, it could be good for planning for not expected for HR apps.